baseplate/integration/wrapped_context.py

```diff
@@ -0,0 +1,21 @@
+import copy
+
+
+def merge_two_dicts(x, y):
+    """Given two dicts, merge them into a new dict as a shallow copy."""
+    z = x.copy()
+    z.update(y)
+    return z
+
+
+class WrappedRequestContext(object):
+    def __init__(self, context, trace=None):
+        self._context = context
+        self.__dict__.update(self._context.__dict__)
+
+    def __getattr__(self, attr):
+        return getattr(self._context, attr)
+
+    def clone(self):
+        new_context = copy.copy(self._context)
```

Comment:
Yeah, I think this was just me running circles around myself. I was worried about the flip side -> innocuous changing of underlying context attributes on accident. I think it's better to keep the reference to existing context object and do some logging every time we access the underlying context instead of changing our wrapped attrs or adding new ones.